<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Perform ordination</title>

<script src="site_libs/header-attrs-2.16/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DietDiveR</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    ASA24
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="ASA24_01_load_and_clean.html">Load and clean ASA24 data</a>
    </li>
    <li>
      <a href="ASA24_02_calc_totals_by_occasion.html">Calculate totals by occasion</a>
    </li>
    <li>
      <a href="ASA24_03_data_overview.html">Data overview</a>
    </li>
    <li>
      <a href="ASA24_04_percent_kcal.html">Percent kcal of CARB, PROT, TFAT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="ASA24_05_prep_clustering.html">Prepare for clustering</a>
    </li>
    <li>
      <a href="ASA24_06_PCA.html">PCA</a>
    </li>
    <li>
      <a href="ASA24_07_k-means.html">k-means</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="ASA24_08_create_foodtree_allfooditems.html">Foodtrees with all food items</a>
    </li>
    <li>
      <a href="ASA24_09_create_foodtree_ownitems.html">Foodtrees with your own data</a>
    </li>
    <li>
      <a href="ASA24_10_viz_food_tree.html">Visualize foodtrees</a>
    </li>
    <li>
      <a href="ASA24_11_ordination.html">Ordination</a>
    </li>
    <li>
      <a href="ASA24_12_viz_ordination.html">Visualize ordination</a>
    </li>
    <li>
      <a href="ASA24_13_corr_axes_foods.html">Correlation between axes and foods</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    NHANES
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="NHANES_01_load_and_clean_1.html">Load and clean NHANES data - 1</a>
    </li>
    <li>
      <a href="NHANES_02_load_and_clean_2.html">Load and clean NHANES data - 2</a>
    </li>
    <li>
      <a href="NHANES_03_add_meta_GLU_index.html">Add metadata and GLU-index</a>
    </li>
    <li>
      <a href="NHANES_04_overview_GLU_index.html">Data overview</a>
    </li>
    <li>
      <a href="NHANES_05_GLU_index_by_age_gender.html">GLU-index by age and gender</a>
    </li>
    <li>
      <a href="NHANES_06_subset_males60to79.html">Subset males 60-79 years old</a>
    </li>
    <li>
      <a href="NHANES_07_males60to79_percent_kcal.html">Percent kcal of CARB, PROT, TFAT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="NHANES_08_males60to79_prep_for_clustering.html">Prepare for clustering</a>
    </li>
    <li>
      <a href="NHANES_09_males60to79_PCA.html">PCA</a>
    </li>
    <li>
      <a href="NHANES_10_males60to79_k-means.html">k-means</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="NHANES_11_males60to79_generate_foodtrees.html">Generate foodtrees</a>
    </li>
    <li>
      <a href="NHANES_12_males60to79_viz_foodtrees.html">Visualize foodtrees</a>
    </li>
    <li>
      <a href="NHANES_13_males60to79_ordination.html">Ordination</a>
    </li>
    <li>
      <a href="NHANES_14_males60to79_viz_ordination.html">Visualize ordination</a>
    </li>
    <li>
      <a href="NHANES_15_males60to79_corr_axes_foods.html">Correlation between axes and foods</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="NHANES_16_Div.html">Assign DivGroups by dietary diversity</a>
    </li>
  </ul>
</li>
<li>
  <a href="Links.html">Links</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Perform ordination</h1>

</div>


<style type="text/css">
h1.title {
  font-size: 38px;
  font-family: "Tahoma", sans-serif;
  color: #1F3600; /* darkgreen from DietDiveR logo */
}
h1 { /* Header 1 */
  font-size: 28px;
  color: #1F3600;
}
h2 { /* Header 2 */
    font-size: 22px;
  color: #1F3600;
}
h3 { /* Header 3 */
  font-size: 18px;
  color: #1F3600;
}
body{ /* Normal  */
  background-color: rgb(251, 251, 251);
  }
blockquote{ 
  font-size: 14px;
}
.list-group-item.active, .list-group-item.active:focus, .list-group-item.active:hover {
    background-color: #1F3600;
}
</style>
<p><br></p>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>In this section, we will take the phylogeny of food items into account in clustering individuals according to their dietary data. In order to do so, we will use <a href="https://joey711.github.io/phyloseq/index.html">the phyloseq package</a>, which uses phylogeny of microbes and their abundance. We will replace microbes with food items consumed by our dietary study participants.</p>
<p><br></p>
</div>
<div id="load-the-necessary-packages-and-functions" class="section level1">
<h1>Load the necessary packages and functions</h1>
<p>Name your main directory for future use.</p>
<pre class="r"><code>main_wd &lt;- &quot;~/GitHub/DietDiveR&quot;</code></pre>
<p><br></p>
<hr />
<p>If you have not downloaded and installed the phyloseq package yet, you can do so by first installing BiocManager (if you have not done so):</p>
<pre class="r"><code>if (!require(&quot;BiocManager&quot;, quietly = TRUE)) install.packages(&quot;BiocManager&quot;)</code></pre>
<p>Then download and install the phyloseq package.</p>
<pre class="r"><code>BiocManager::install(&quot;phyloseq&quot;)</code></pre>
<p>Install the devtools package necessary for installing the pairwiseAdonis package.</p>
<pre class="r"><code>if (!require(&quot;devtools&quot;,    quietly = TRUE))install.packages(&quot;devtools&quot;)</code></pre>
<pre><code>## Warning: package &#39;devtools&#39; was built under R version 4.1.3</code></pre>
<pre><code>## Warning: package &#39;usethis&#39; was built under R version 4.1.3</code></pre>
<pre><code>## 
## Attaching package: &#39;devtools&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:BiocManager&#39;:
## 
##     install</code></pre>
<p>Install pairwise adonis function from Github. (<a href="https://github.com/pmartinezarbizu/pairwiseAdonis" class="uri">https://github.com/pmartinezarbizu/pairwiseAdonis</a>)</p>
<pre class="r"><code>devtools::install_github(&quot;pmartinezarbizu/pairwiseAdonis/pairwiseAdonis&quot;)</code></pre>
<p><br></p>
<hr />
<p>load the necessary packages.</p>
<pre class="r"><code>library(phyloseq)</code></pre>
<pre><code>## 
## Attaching package: &#39;phyloseq&#39;</code></pre>
<pre><code>## The following object is masked _by_ &#39;.GlobalEnv&#39;:
## 
##     distance</code></pre>
<pre class="r"><code>library(ggtree)
library(pairwiseAdonis)</code></pre>
<pre><code>## Loading required package: vegan</code></pre>
<pre><code>## Loading required package: permute</code></pre>
<pre><code>## 
## Attaching package: &#39;permute&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:devtools&#39;:
## 
##     check</code></pre>
<pre><code>## Loading required package: lattice</code></pre>
<pre><code>## This is vegan 2.5-7</code></pre>
<p>Load necessary functions and ggplot formatting themes.</p>
<pre class="r"><code>source(&quot;lib/specify_data_dir.R&quot;)
source(&quot;lib/ordination.R&quot;)
source(&quot;lib/ggplot2themes.R&quot;)
source(&quot;lib/sort_IFC_by_ID.R&quot;)
source(&quot;lib/plot.axis.1to4.by.factor.R&quot;)</code></pre>
<p>Load the distinct 100 colors for use.</p>
<pre class="r"><code>distinct100colors &lt;- readRDS(&quot;lib/distinct100colors.rda&quot;)  </code></pre>
<p>You can come back to the main directory by:</p>
<pre class="r"><code>setwd(main_wd)</code></pre>
<p><br></p>
</div>
<div id="create-a-phyloseq-object-for-ordination" class="section level1">
<h1>Create a phyloseq object for ordination</h1>
<p>Specify the directory where the data is.</p>
<pre class="r"><code>SpecifyDataDirectory(directory.name = &quot;eg_data/VVKAJ/&quot;)</code></pre>
<pre><code>## [1] &quot;The data directory has been set as&quot;</code></pre>
<pre><code>## [1] &quot;~/GitHub/DietDiveR/eg_data/VVKAJ/&quot;</code></pre>
<!-- This is the R-markdown code to change directories. -->
<p><br></p>
<div id="load-files-for-creating-a-phyloseq-object" class="section level2">
<h2>Load files for creating a phyloseq object</h2>
<div id="food" class="section level3">
<h3>Food</h3>
<p>Load IFC table, and sort the columnnames (userID), leaving the last column (taxonomy) intact. This dataframe will be saved as <code>food</code>. Also, save <code>food</code> as a .txt file to be used in the “correlation between Axes and foods” section.</p>
<pre class="r"><code>SortIFCByID(ifc.input =           &quot;Foodtree/VVKAJ_Items_f_id_s_m_QCed_4Lv.food.ifc.txt&quot;,
            outfn.for.corr.axis = &quot;Foodtree/VVKAJ_Items_f_id_s_m_QCed_4Lv.food.ifc_sorted.txt&quot;)</code></pre>
<p><code>food</code> is a matrix of Food descriptions (rows) x SampleID (columns).</p>
<pre class="r"><code>head(food)[1:6, 1:4]</code></pre>
<pre><code>##                       vvkaj.00001 vvkaj.00002 vvkaj.00003 vvkaj.00004
## Milk NFS                        0           0           0           0
## Milk whole                      0         183           0           0
## Milk reduced fat 2              0           0           0           0
## Milk low fat 1                  0           0          61           0
## Soy milk                        0           0           0           0
## Almond milk sweetened           0           0           0           0</code></pre>
<p>Format the food object and create an ifc_table called <code>IFC</code>.</p>
<pre class="r"><code>PrepFood(data= food)</code></pre>
</div>
<div id="taxonomy-tax" class="section level3">
<h3>Taxonomy (tax)</h3>
<pre class="r"><code>tax &lt;- read.delim(&quot;Foodtree/VVKAJ_Items_f_id_s_m_QCed_4Lv.tax.txt&quot;)</code></pre>
<p>Format the tax file and create a taxonomy table called <code>TAX</code>.</p>
<pre class="r"><code>PrepTax(data= tax)</code></pre>
</div>
<div id="sample" class="section level3">
<h3>Sample</h3>
<pre class="r"><code>meta &lt;- read.table( &quot;ind_metadata_UserxDay.txt&quot;, sep=&quot;\t&quot;, header=T)</code></pre>
<p>Format the metadata file and save it as <code>SAMPLES</code>.</p>
<pre class="r"><code>PrepMeta(data= meta)</code></pre>
</div>
<div id="foodtree" class="section level3">
<h3>Foodtree</h3>
<pre class="r"><code>foodtree &lt;- read_tree(&quot;Foodtree/VVKAJ_Items_f_id_s_m_QCed_4Lv.tree.nwk&quot;)</code></pre>
<p>It is OK to see a message saying that<br />
“Found more than one class”phylo” in cache; using the first, from namespace ‘phyloseq’<br />
Also defined by ‘tidytree’“.</p>
<p>Format foodtree and save it as <code>TREE</code>.</p>
<pre class="r"><code>PrepTree(data= foodtree)</code></pre>
<p>Again, it is OK to see the same message as the previous line.</p>
<p><br></p>
<hr />
</div>
</div>
<div id="create-a-phyloseq-object-with-ifc-tax-samples-and-tree" class="section level2">
<h2>Create a phyloseq object with IFC, TAX, SAMPLES, and TREE</h2>
<pre class="r"><code>phyfoods &lt;- phyloseq(IFC, TAX, SAMPLES, TREE)</code></pre>
<p>It is OK to see a message (or multiple of them) saying that<br />
Found more than one class “phylo” in cache; using the first, from namespace ‘phyloseq’<br />
Also defined by ‘tidytree’.</p>
<p><br></p>
<div id="check-your-metadata" class="section level3">
<h3>Check your metadata</h3>
<p>Show the sample names and ensure they are vvkaj.00xxx.</p>
<pre class="r"><code>sample_names(phyfoods)</code></pre>
<pre><code>##  [1] &quot;vvkaj.00001&quot; &quot;vvkaj.00002&quot; &quot;vvkaj.00003&quot; &quot;vvkaj.00004&quot; &quot;vvkaj.00005&quot;
##  [6] &quot;vvkaj.00006&quot; &quot;vvkaj.00007&quot; &quot;vvkaj.00008&quot; &quot;vvkaj.00009&quot; &quot;vvkaj.00010&quot;
## [11] &quot;vvkaj.00011&quot; &quot;vvkaj.00012&quot; &quot;vvkaj.00013&quot; &quot;vvkaj.00014&quot; &quot;vvkaj.00015&quot;
## [16] &quot;vvkaj.00016&quot; &quot;vvkaj.00017&quot; &quot;vvkaj.00018&quot; &quot;vvkaj.00019&quot; &quot;vvkaj.00020&quot;
## [21] &quot;vvkaj.00021&quot; &quot;vvkaj.00022&quot; &quot;vvkaj.00023&quot; &quot;vvkaj.00024&quot; &quot;vvkaj.00025&quot;
## [26] &quot;vvkaj.00026&quot; &quot;vvkaj.00027&quot; &quot;vvkaj.00028&quot; &quot;vvkaj.00029&quot; &quot;vvkaj.00030&quot;
## [31] &quot;vvkaj.00031&quot; &quot;vvkaj.00032&quot; &quot;vvkaj.00033&quot; &quot;vvkaj.00034&quot; &quot;vvkaj.00035&quot;
## [36] &quot;vvkaj.00036&quot; &quot;vvkaj.00037&quot; &quot;vvkaj.00038&quot; &quot;vvkaj.00039&quot; &quot;vvkaj.00040&quot;
## [41] &quot;vvkaj.00041&quot; &quot;vvkaj.00042&quot; &quot;vvkaj.00043&quot; &quot;vvkaj.00044&quot; &quot;vvkaj.00045&quot;</code></pre>
<p>Show metadata.</p>
<pre class="r"><code>head(sample_data(phyfoods), n=3)</code></pre>
<pre><code>##                SampleID UserName StudyDayNo       Diet Gender Age Weight Height
## vvkaj.00001 vvkaj.00001 VVKAJ101          1 Vegetarian      M  31     79    186
## vvkaj.00002 vvkaj.00002 VVKAJ101          2 Vegetarian      M  31     79    186
## vvkaj.00003 vvkaj.00003 VVKAJ101          3 Vegetarian      M  31     79    186
##                  BMI Waist.Circumference
## vvkaj.00001 22.83501                  80
## vvkaj.00002 22.83501                  80
## vvkaj.00003 22.83501                  80</code></pre>
<p>Check the level 1 foods in your food tree.</p>
<pre class="r"><code>L1s = tax_table(phyfoods)[, &quot;L1&quot;]
as.vector(unique(L1s))</code></pre>
<pre><code>## [1] &quot;L1_Milk_and_Milk_Products&quot;                     
## [2] &quot;L1_Meat_Poultry_Fish_and_Mixtures&quot;             
## [3] &quot;L1_Eggs&quot;                                       
## [4] &quot;L1_Dry_Beans_Peas_Other_Legumes_Nuts_and_Seeds&quot;
## [5] &quot;L1_Grain_Product&quot;                              
## [6] &quot;L1_Fruits&quot;                                     
## [7] &quot;L1_Vegetables&quot;                                 
## [8] &quot;L1_Fats_Oils_and_Salad_Dressings&quot;              
## [9] &quot;L1_Sugars_Sweets_and_Beverages&quot;</code></pre>
<p><br></p>
</div>
</div>
</div>
<div id="perform-ordination-with-weighted-unifrac-distance" class="section level1">
<h1>Perform ordination with weighted unifrac distance</h1>
<p>Change to the folder called “Ordination” in your “VVKAJ” folder.</p>
<pre class="r"><code>SpecifyDataDirectory(directory.name = &quot;eg_data/VVKAJ/Ordination/&quot;)</code></pre>
<!-- This is the R-markdown code to change directories. -->
<p>Perform Principal Coordinate Analysis (PCoA) with weighted unifrac distance of your food data. Ordination by UNweighted unifrac distances can be done by having the “weighted” argument as FALSE. This may take a few minutes depending on your data size. e.g. a large phyloseq object (7.9 MB) takes ~ 1 min.</p>
<pre class="r"><code>ordinated &lt;- phyloseq::ordinate(phyfoods, method=&quot;PCoA&quot;, distance=&quot;unifrac&quot;, weighted=TRUE) </code></pre>
<p>Save the percent variance explained as a txt file.</p>
<pre class="r"><code>Eigen(eigen.input = ordinated$values$Relative_eig, 
      output.fn=&quot;VVKAJ_Items_f_id_s_m_QCed_4Lv_ord_WEIGHTED_eigen_percent.txt&quot;)</code></pre>
<p>Merge the first n axes to the metadata and save it as a txt file. This will be used for plotting ordination results.</p>
<pre class="r"><code>MergeAxesAndMetadata(ord.object=ordinated, number.of.axes=10, meta.data= meta, 
                     output.fn= &quot;VVKAJ_Items_f_id_s_m_QCed_4Lv_ord_WEIGHTED_meta_users.txt&quot;)</code></pre>
<p><br></p>
<div id="plot-your-ordination-results" class="section level2">
<h2>Plot your ordination results</h2>
<p>Read in the eigenvalues for axis labels of biplots.</p>
<pre class="r"><code>eigen_loaded &lt;- read.table(&quot;VVKAJ_Items_f_id_s_m_QCed_4Lv_ord_WEIGHTED_eigen_percent.txt&quot;, header=T)</code></pre>
<p>Make a vector that contains the variance explained.</p>
<pre class="r"><code>eigen_loaded_vec &lt;- eigen_loaded[, 2]</code></pre>
<p>Read in the metadata and users’ Axis values.</p>
<pre class="r"><code>meta_usersdf &lt;- read.table(&quot;VVKAJ_Items_f_id_s_m_QCed_4Lv_ord_WEIGHTED_meta_users.txt&quot;, header=T)    </code></pre>
<p>Change Diet to a factor so that factor levels will be displayed in order.</p>
<pre class="r"><code>meta_usersdf$Diet &lt;- factor(meta_usersdf$Diet,
                            levels= c(&quot;Vegetarian&quot;, &quot;Vegan&quot;, &quot;Keto&quot;, &quot;American&quot;, &quot;Japanese&quot;))  </code></pre>
<p>Take a look at meta_usersdf that has been loaded.</p>
<pre class="r"><code>head(meta_usersdf, 3)</code></pre>
<pre><code>##     Row.names    SampleID UserName StudyDayNo       Diet Gender Age Weight
## 1 vvkaj.00001 vvkaj.00001 VVKAJ101          1 Vegetarian      M  31     79
## 2 vvkaj.00002 vvkaj.00002 VVKAJ101          2 Vegetarian      M  31     79
## 3 vvkaj.00003 vvkaj.00003 VVKAJ101          3 Vegetarian      M  31     79
##   Height      BMI Waist.Circumference        Axis.1      Axis.2      Axis.3
## 1    186 22.83501                  80 -0.0442801041  0.02759311 -0.01355771
## 2    186 22.83501                  80 -0.0007933678 -0.04075574  0.07943972
## 3    186 22.83501                  80 -0.0842314600  0.02669820 -0.02079188
##        Axis.4      Axis.5      Axis.6      Axis.7      Axis.8      Axis.9
## 1 -0.04497503  0.06643215 -0.04633301  0.03528304 -0.03953240  0.04564185
## 2 -0.02767431 -0.05829929  0.02566912 -0.01358921 -0.03447211 -0.11926416
## 3 -0.10074383  0.03985384  0.01108057 -0.02331700 -0.03071878 -0.04394001
##       Axis.10
## 1  0.03792176
## 2 -0.04040412
## 3 -0.03060788</code></pre>
<p><br></p>
<hr />
<p>Save Axes 1 &amp; 2, 1 &amp; 3, 2 &amp; 3, 3 &amp; 4, 2 &amp; 4 biplots with and without ellipses with specified confidence interval.</p>
<p>The results are saved with filenames with the specified “prefix_AxisXY.pdf” or “prefix_AxisXY_ellipses.pdf”. You need to supply the same number of colors in the order of the factor level to be used. <code>dot.colors</code> are for datapoints, and <code>ellipses.colors</code> are for ellipses outlines.</p>
<pre class="r"><code>PlotAxis1to4ByFactor(axis.meta.df    = meta_usersdf, 
                     factor.to.color = &quot;Diet&quot;, 
                     eigen.vector    = eigen_loaded_vec,
                     dot.colors      = distinct100colors, 
                     ellipses.colors = distinct100colors,  
                     ellipses.cflevel = 0.95,
                     out.prefix = &quot;VVKAJ_Items_f_id_s_m_QCed_4Lv_ord_WEIGHTED_diet&quot;
                     )</code></pre>
<p><br></p>
<p>The function above saves 10 biplots in the PDF format. Let us take a look at Axis 1 vs. Axis 2.</p>
<p><img src="Screenshots_for_Tutorial/ASA24%2011%20ordination/4Lv%20WEIGHTED%20ord.png" style="width:60.0%" /></p>
<p>It seems that Axis1 separated the Japanese diet group from the rest, and Axis2 separated the Keto diet group from the rest.</p>
<p><br></p>
<hr />
</div>
<div id="permanova-tests" class="section level2">
<h2>Permanova tests</h2>
<p>Some of the Diet groups seem to form distinct clusters. Use beta-diversity and adonis (permanova) tests to see if they are actually distinct from one another.</p>
<p>Generate a weighted unifrac distance matrix.</p>
<pre class="r"><code>dist_matrix &lt;- phyloseq::distance(phyfoods, method = &quot;wunifrac&quot;)</code></pre>
<p>Perform dispersion test.</p>
<p>vegan::betadisper computes centeroids and distance of each datapoint from it.</p>
<pre class="r"><code>dispr &lt;- vegan::betadisper(d=dist_matrix, phyloseq::sample_data(phyfoods)$Diet)</code></pre>
<p>Show the centroids and dispersion of each group.</p>
<pre class="r"><code>plot(dispr)</code></pre>
<p><img src="ASA24_11_ordination_files/figure-html/unnamed-chunk-36-1.png" width="70%" /></p>
<p>Use dispr to do a permutation test for homogeneity of multivariate dispersion. The set.seed function ensures the same permutation results will be obtained every time; otherwise, the p-values will slightly differ each run, as it is a permutation test.</p>
<pre class="r"><code>set.seed(123)
vegan::permutest(dispr, perm=5000)</code></pre>
<pre><code>## 
## Permutation test for homogeneity of multivariate dispersions
## Permutation: free
## Number of permutations: 5000
## 
## Response: Distances
##           Df   Sum Sq   Mean Sq      F N.Perm  Pr(&gt;F)  
## Groups     4 0.022031 0.0055078 2.6938   5000 0.04839 *
## Residuals 40 0.081785 0.0020446                        
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>If p&gt;0.05, the dispersion of each group are not different, and the assumption for adonis is met. The results here indicate that the dispersion of each group may be different, so we should consider this information in discussion. Nevertheless, we will proceed for demonstration purposes.</p>
<p>Use adonis to test whether there is a difference between groups’ composition. i.e., composition among groups (food they consumed) is similar or not.</p>
<pre class="r"><code>set.seed(123)
vegan::adonis(dist_matrix ~ phyloseq::sample_data(phyfoods)$Diet, permutations = 5000)</code></pre>
<pre><code>## 
## Call:
## vegan::adonis(formula = dist_matrix ~ phyloseq::sample_data(phyfoods)$Diet,      permutations = 5000) 
## 
## Permutation: free
## Number of permutations: 5000
## 
## Terms added sequentially (first to last)
## 
##                                      Df SumsOfSqs  MeanSqs F.Model      R2
## phyloseq::sample_data(phyfoods)$Diet  4    0.9401 0.235020  3.4251 0.25513
## Residuals                            40    2.7447 0.068617         0.74487
## Total                                44    3.6848                  1.00000
##                                      Pr(&gt;F)    
## phyloseq::sample_data(phyfoods)$Diet  2e-04 ***
## Residuals                                      
## Total                                          
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>If overall adonis is significant, which is true in this case, you can run pairwise adonis to see which group pairs are different.</p>
<pre class="r"><code>pairwise.adonis(dist_matrix, phyloseq::sample_data(phyfoods)$Diet, perm = 5000,
                p.adjust.m = &quot;none&quot;)    </code></pre>
<pre><code>##                     pairs Df  SumsOfSqs  F.Model         R2    p.value
## 1     Vegetarian vs Vegan  1 0.07557735 1.178911 0.06862547 0.20815837
## 2      Vegetarian vs Keto  1 0.13097005 1.888662 0.10557872 0.00739852
## 3  Vegetarian vs American  1 0.15189915 2.540675 0.13703252 0.00019996
## 4  Vegetarian vs Japanese  1 0.31482162 5.654587 0.26112652 0.00039992
## 5           Vegan vs Keto  1 0.19900379 2.430678 0.13188217 0.00019996
## 6       Vegan vs American  1 0.16773768 2.319599 0.12661843 0.00019996
## 7       Vegan vs Japanese  1 0.36815946 5.398094 0.25226984 0.00019996
## 8        Keto vs American  1 0.21818296 2.813418 0.14954314 0.00019996
## 9        Keto vs Japanese  1 0.32249324 4.391284 0.21535104 0.00019996
## 10   American vs Japanese  1 0.40135753 6.282904 0.28196075 0.00019996
##    p.adjusted sig
## 1  0.20815837    
## 2  0.00739852   *
## 3  0.00019996  **
## 4  0.00039992  **
## 5  0.00019996  **
## 6  0.00019996  **
## 7  0.00019996  **
## 8  0.00019996  **
## 9  0.00019996  **
## 10 0.00019996  **</code></pre>
<p><br></p>
<p>This table indicates all the combination of the five diets are significantly different (p&lt;0.05) except Vegetarian vs. Vegan. We expect Vegetarian and Vegan diets to be similar, so this makes sense.</p>
<p><br></p>
<hr />
</div>
<div id="save-weighted-unifrac-distance-matrix" class="section level2">
<h2>Save weighted unifrac distance matrix</h2>
<p>Generate and save a weighted unifrac distance matrix of “Samples”.</p>
<pre class="r"><code>WeightedUnifracDis(input.phyloseq.obj = phyfoods, 
                   output.fn = &quot;VVKAJ_Items_f_id_s_m_QCed_4Lv_WEIGHTED_uni_dis.txt&quot;)        </code></pre>
<p><br></p>
<hr />
</div>
</div>
<div id="perform-ordination-with-unweighted-unifrac-distance" class="section level1">
<h1>Perform ordination with unweighted unifrac distance</h1>
<p>You can perform Principal Coordinate Analysis (PCoA) with UNweighted unifrac distance of your food data.</p>
<pre class="r"><code>ordinated_u = phyloseq::ordinate(phyfoods, method=&quot;PCoA&quot;, distance=&quot;unifrac&quot;, weighted=FALSE)  </code></pre>
<p>Use the same code above for creating plots, but now with ordinated_u for the ord.object argument, and change WEIGHTED to UNweighted, or an appropriate name for the method you selected.</p>
<p>Save the percent variance explained as a txt file.</p>
<pre class="r"><code>Eigen(eigen.input = ordinated_u$values$Relative_eig, 
      output.fn=&quot;VVKAJ_Items_f_id_s_m_QCed_4Lv_ord_UNweighted_eigen_percent.txt&quot;)</code></pre>
<p>Merge the first n axes to the metadata and save it as a txt file. This will be used for plotting ordination results.</p>
<pre class="r"><code>MergeAxesAndMetadata(ord.object=ordinated_u, number.of.axes=10, meta.data= meta, 
                     output.fn= &quot;VVKAJ_Items_f_id_s_m_QCed_4Lv_ord_UNweighted_meta_users.txt&quot;)</code></pre>
<p><br></p>
<hr />
<div id="plot-your-ordination-results-1" class="section level2">
<h2>Plot your ordination results</h2>
<p>Read in the eigenvalues for axis labels of biplots.</p>
<pre class="r"><code>eigen_loaded &lt;- read.table(&quot;VVKAJ_Items_f_id_s_m_QCed_4Lv_ord_UNweighted_eigen_percent.txt&quot;, header=T)</code></pre>
<p>Make a vector that contains the variance explained.</p>
<pre class="r"><code>eigen_loaded_vec &lt;- eigen_loaded[, 2]</code></pre>
<p>Read in the metadata and users’ Axis values.</p>
<pre class="r"><code>meta_usersdf &lt;- read.table(&quot;VVKAJ_Items_f_id_s_m_QCed_4Lv_ord_UNweighted_meta_users.txt&quot;, header=T)    </code></pre>
<p>Change Diet to a factor so that factor levels will be displayed in order.</p>
<pre class="r"><code>meta_usersdf$Diet &lt;- factor(meta_usersdf$Diet,
                            levels= c(&quot;Vegetarian&quot;, &quot;Vegan&quot;, &quot;Keto&quot;, &quot;American&quot;, &quot;Japanese&quot;))  </code></pre>
<p>Take a look at meta_usersdf that has been loaded.</p>
<pre class="r"><code>head(meta_usersdf, 3)</code></pre>
<pre><code>##     Row.names    SampleID UserName StudyDayNo       Diet Gender Age Weight
## 1 vvkaj.00001 vvkaj.00001 VVKAJ101          1 Vegetarian      M  31     79
## 2 vvkaj.00002 vvkaj.00002 VVKAJ101          2 Vegetarian      M  31     79
## 3 vvkaj.00003 vvkaj.00003 VVKAJ101          3 Vegetarian      M  31     79
##   Height      BMI Waist.Circumference      Axis.1      Axis.2      Axis.3
## 1    186 22.83501                  80 -0.09008072  0.12526150 0.224690714
## 2    186 22.83501                  80 -0.10016605 -0.29241014 0.008973829
## 3    186 22.83501                  80 -0.17290956 -0.04453934 0.065796514
##        Axis.4       Axis.5      Axis.6     Axis.7      Axis.8      Axis.9
## 1 -0.09324589  0.008335857 -0.08133357 -0.1488169  0.02877081 -0.06084911
## 2 -0.19562671 -0.091767769  0.08512958 -0.1743513  0.05350784 -0.03272327
## 3 -0.18999341  0.134560378  0.14815162 -0.1078155 -0.13799218  0.25383971
##       Axis.10
## 1  0.11004841
## 2 -0.04213860
## 3  0.03525768</code></pre>
<p><br></p>
<hr />
<p>Save Axes 1 &amp; 2, 1 &amp; 3, 2 &amp; 3, 3 &amp; 4, 2 &amp; 4 biplots with and without ellipses with specified confidence interval. The results are saved with filenames with the specified “prefix_AxisXY.pdf” or “prefix_AxisXY_ellipses.pdf”. You need to supply the same number of colors in the order of the factor level to be used. <code>dot.colors</code> are for datapoints, and <code>ellipses.colors</code> are for ellipses outlines.</p>
<pre class="r"><code>PlotAxis1to4ByFactor(axis.meta.df    = meta_usersdf, 
                     factor.to.color = &quot;Diet&quot;, 
                     eigen.vector    = eigen_loaded_vec,
                     dot.colors      = distinct100colors, 
                     ellipses.colors = distinct100colors,  
                     ellipses.cflevel = 0.95,
                     out.prefix = &quot;VVKAJ_Items_f_id_s_m_QCed_4Lv_ord_UNweighted_diet&quot;
                     )</code></pre>
<p><br></p>
<hr />
</div>
<div id="permanova-tests-1" class="section level2">
<h2>Permanova tests</h2>
<p>Some of the Diet groups seem to form distinct clusters. Use beta-diversity and adonis tests to see if they are actually distinct from one another.</p>
<p>Generate a weighted unifrac distance matrix.</p>
<pre class="r"><code>dist_matrix &lt;- phyloseq::distance(phyfoods, method = &quot;unifrac&quot;)  # UNweighted</code></pre>
<p>Dispersion test and plot<br />
vegan::betadisper computes centeroids and distance of each datapoint from it.</p>
<pre class="r"><code>dispr &lt;- vegan::betadisper(d=dist_matrix, phyloseq::sample_data(phyfoods)$Diet)</code></pre>
<p>Show the centroids and dispersion of each group.</p>
<pre class="r"><code>plot(dispr)</code></pre>
<p><img src="ASA24_11_ordination_files/figure-html/unnamed-chunk-52-1.png" width="70%" /></p>
<p>Use dispr to do a permutation test for homogeneity of multivariate dispersion.</p>
<pre class="r"><code>set.seed(123)
vegan::permutest(dispr, perm=5000)</code></pre>
<pre><code>## 
## Permutation test for homogeneity of multivariate dispersions
## Permutation: free
## Number of permutations: 5000
## 
## Response: Distances
##           Df   Sum Sq   Mean Sq     F N.Perm Pr(&gt;F)    
## Groups     4 0.047486 0.0118715 10.91   5000  2e-04 ***
## Residuals 40 0.043525 0.0010881                        
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>If p&gt;0.05, the dispersion of each group are not different, and the assumption for adonis is met. The results here indicate that the dispersion of each group may be different, so we should consider this information in discussion. Nevertheless, we will proceed for demonstration purposes.</p>
<p>Use adonis to test whether there is a difference between groups’ composition. i.e., composition among groups (food they consumed) is similar or not.</p>
<pre class="r"><code>set.seed(123)
vegan::adonis(dist_matrix ~ phyloseq::sample_data(phyfoods)$Diet, permutations = 5000)</code></pre>
<pre><code>## 
## Call:
## vegan::adonis(formula = dist_matrix ~ phyloseq::sample_data(phyfoods)$Diet,      permutations = 5000) 
## 
## Permutation: free
## Number of permutations: 5000
## 
## Terms added sequentially (first to last)
## 
##                                      Df SumsOfSqs MeanSqs F.Model      R2
## phyloseq::sample_data(phyfoods)$Diet  4     3.696 0.92399  2.2278 0.18219
## Residuals                            40    16.590 0.41475         0.81781
## Total                                44    20.286                 1.00000
##                                      Pr(&gt;F)    
## phyloseq::sample_data(phyfoods)$Diet  2e-04 ***
## Residuals                                      
## Total                                          
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>If overall adonis is significant, which is true in this case, you can run pairwise adonis to see which group pairs are different.</p>
<pre class="r"><code>pairwise.adonis(dist_matrix, phyloseq::sample_data(phyfoods)$Diet, perm = 5000,
                p.adjust.m = &quot;none&quot;)    </code></pre>
<pre><code>##                     pairs Df SumsOfSqs   F.Model         R2    p.value
## 1     Vegetarian vs Vegan  1 0.4374127 0.9882078 0.05817022 0.50789842
## 2      Vegetarian vs Keto  1 0.6542830 1.5148347 0.08648867 0.00139972
## 3  Vegetarian vs American  1 0.7838799 1.7984651 0.10104608 0.00019996
## 4  Vegetarian vs Japanese  1 1.1927048 3.1013470 0.16236274 0.00039992
## 5           Vegan vs Keto  1 0.7691534 1.7648633 0.09934573 0.00059988
## 6       Vegan vs American  1 0.7535836 1.7136319 0.09674085 0.00019996
## 7       Vegan vs Japanese  1 1.2403751 3.1929421 0.16636022 0.00019996
## 8        Keto vs American  1 0.9196267 2.1434374 0.11813844 0.00019996
## 9        Keto vs Japanese  1 1.1584619 3.0666706 0.16083933 0.00019996
## 10   American vs Japanese  1 1.3304323 3.4855237 0.17887760 0.00019996
##    p.adjusted sig
## 1  0.50789842    
## 2  0.00139972   *
## 3  0.00019996  **
## 4  0.00039992  **
## 5  0.00059988  **
## 6  0.00019996  **
## 7  0.00019996  **
## 8  0.00019996  **
## 9  0.00019996  **
## 10 0.00019996  **</code></pre>
<p>Similarly to the results with weighted unifrac distances, this table also indicates all the combination of the five diets are significantly different (p&lt;0.05) except Vegetarian vs. Vegan. We expect Vegetarian and Vegan diets to be similar, so this makes sense.</p>
<p><br></p>
<hr />
</div>
<div id="save-unweighted-unifrac-distance-matrix" class="section level2">
<h2>Save unweighted unifrac distance matrix</h2>
<p>Generate and save an unweighted unifrac distance matrix of “Samples”.</p>
<pre class="r"><code>UnweightedUnifracDis(input.phyloseq.obj = phyfoods, 
                     output.fn = &quot;VVKAJ_Items_f_id_s_m_QCed_4Lv_UNweighted_uni_dis.txt&quot;)        </code></pre>
<p><br></p>
<hr />
<p>Come back to the main directory before you start running another script.</p>
<pre class="r"><code>setwd(main_wd)</code></pre>
<!-- This is to remove big white space at the end of the rendered html, which is produced due to toc_floating.  Source: https://stackoverflow.com/questions/52933437/how-to-remove-white-space-at-the-end-of-an-rmarkdown-html-output -->
<div class="tocify-extend-page" data-unique="tocify-extend-page" style="height: 0;">

</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
